webapi/src/main/scala/org/knora/webapi/twirl/StandoffTag.scala

```diff
@@ -0,0 +1,163 @@
+import org.knora.webapi.util.KnoraIdUtil
+
+/**
+  * The following case classes represent standoff classes and attributes
+  * that are about to be added to the triplestore.
+  *
+  */
+
+
+/*
+    trait for standoff tag attribute
+ */
+trait StandoffTagAttributeV1 {
+
+    def standoffPropertyIri: IRI
+
+    def stringValue: String
+
+    def rdfValue: String
+
+}
+
+/**
+  * Represents a standoff tag attribute of type Iri.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagIriAttributeV1(standoffPropertyIri: IRI, value: IRI) extends StandoffTagAttributeV1 {
+
+    def stringValue = value
+
+    def rdfValue = s"<$value>"
+
+}
+
+/**
+  * Represents a standoff tag attribute that refers to another standoff node.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagInternalReferenceAttributeV1(standoffPropertyIri: IRI, value: String) extends StandoffTagAttributeV1 {
+
+    def stringValue = value.toString
+
+    def rdfValue = s"<$value>"
+
+}
+
+/**
+  * Represents a standoff tag attribute of type string.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagStringAttributeV1(standoffPropertyIri: IRI, value: String) extends StandoffTagAttributeV1 {
+
+    def stringValue = value
+
+    def rdfValue = s"""\"\"\"$value\"\"\""""
+
+}
+
+/**
+  * Represents a standoff tag attribute of type integer.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagIntegerAttributeV1(standoffPropertyIri: IRI, value: Int) extends StandoffTagAttributeV1 {
+
+    def stringValue = value.toString
+
+    def rdfValue = value.toString
+
+}
+
+/**
+  * Represents a standoff tag attribute of type decimal.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagDecimalAttributeV1(standoffPropertyIri: IRI, value: BigDecimal) extends StandoffTagAttributeV1 {
+
+    def stringValue = value.toString
+
+    def rdfValue = s""""${value.toString}"^^xsd:decimal"""
+
+}
+
+/**
+  * Represents a standoff tag attribute of type boolean.
+  *
+  * @param standoffPropertyIri the Iri of the standoff property
+  * @param value the value of the standoff property.
+  */
+case class StandoffTagBooleanAttributeV1(standoffPropertyIri: IRI, value: Boolean) extends StandoffTagAttributeV1 {
+
+    def stringValue = value.toString
+
+    def rdfValue = value.toString
+
+}
+
+/**
+  * Represents a `knora-base:StandoffTag` or any subclass of it
+  * that is not a subclass of a data type standoff tag (e.g. `knora-base:StandoffDateTag`).
```

Comment:
You are right, this comment does not make any sense.

I think I put it there when I still wanted to have one case class per data type. But then we decided that the data type specifics go into the attributes so the standoff tags can be handled generically in the SPARQL Update templates.